### PR TITLE
Move truffleruby/src to src

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,6 +38,7 @@ test/truffle/cexts/**/mkmf.log
 
 # Maven
 lib/truffleruby.jar
+target
```
Comment (eregon) [inline .gitignore]: This should be `/target` to match only at the root and not in any sub-directory.

File: doc/contributor/raw-commands.md [hunk 2]
```diff
@@ -23,8 +23,8 @@ to use the `-J-cmd` option.
 
 ```
 >>> jt run -J-cmd -e 'puts "hello"'
-$ /Users/chrisseaton/Documents/ruby/truffleruby/bin/truffleruby -Xcore.load_path=/Users/chrisseaton/Documents/ruby/truffleruby/truffleruby/src/main/ruby -Xgraal.warn_unless=false -J-cmd -e 'puts "hello"'
-$ /Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/bin/java -Dfile.encoding=UTF-8 -Xbootclasspath/a:/Users/chrisseaton/Documents/ruby/truffleruby/lib/truffleruby.jar org.truffleruby.Main -Xhome=/Users/chrisseaton/Documents/ruby/truffleruby -Xlauncher=/Users/chrisseaton/Documents/ruby/truffleruby/bin/truffleruby -Xcore.load_path=/Users/chrisseaton/Documents/ruby/truffleruby/truffleruby/src/main/ruby -Xgraal.warn_unless=false -e puts "hello"
+$ /Users/chrisseaton/Documents/ruby/truffleruby/bin/truffleruby -Xcore.load_path=/Users/chrisseaton/Documents/ruby/src/main/ruby -Xgraal.warn_unless=false -J-cmd -e 'puts "hello"'
```
Comment (eregon) [inline doc/contributor/raw-commands.md]: This and below removes one level too much.

File: .gitignore (hunk 1 above)
Comment (chrisseaton) [inline .gitignore]: Ok will change. But are most of the rest wrong as well then? `lib/truffleruby.jar` for example?

Comment (eregon) [inline .gitignore]: When there is a `/` it implicitly only looks at the root of the repo.

Comment (eregon) [inline .gitignore]: It would probably be more consistent to have a leading`/` in those cases with `/` in the middle as well as it causes no harm.

**Pull request merged**